src/Formats/FormatFactory.cpp

```diff
@@ -191,7 +191,8 @@ InputFormatPtr FormatFactory::getInput(
 
 
         ParallelParsingInputFormat::Params params{
-            buf, sample, parser_creator, file_segmentation_engine, name, settings.max_threads, settings.min_chunk_bytes_for_parallel_parsing};
+            buf, sample, parser_creator, file_segmentation_engine, name, settings.max_threads, settings.min_chunk_bytes_for_parallel_parsing,
+               context->getApplicationType() == Context::ApplicationType::SERVER};
```

Comment:
But there can be the same problem in local?

Reply:
btw I tried your query from pr description and it did not print exception twice (tried both interactive and non-interactive)